x/ibc/core/03-connection/keeper/handshake_test.go

```diff
@@ -773,3 +782,60 @@ func (suite *KeeperTestSuite) TestConnOpenConfirm() {
 		})
 	}
 }
+
+// Ensure consensus params are correctly validated by executing messages. Consensus params are
+// set in context by baseapp so test should deliver messages and not call the functions directly.
+// Only invalid cases are tested since successful instances are by default tested by the testing
+// package.
+func (suite *KeeperTestSuite) TestConsensusParamsValidation() {
+	// invalid client state in ConnOpenTry
+	clientA, clientB := suite.coordinator.SetupClients(suite.chainA, suite.chainB, ibctesting.Tendermint)
+	connA, connB, err := suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
+	suite.Require().NoError(err)
+
+	// set incorrect consensus params on counterparty client on chainA
+	clientState := suite.chainA.GetClientState(clientA)
+	tmClient, ok := clientState.(*ibctmtypes.ClientState)
+	suite.Require().True(ok)
+	tmClient.ConsensusParams.Evidence.MaxAgeDuration++
+
+	suite.chainA.App.IBCKeeper.ClientKeeper.SetClientState(suite.chainA.GetContext(), clientA, tmClient)
+
+	// should fail on validate self client
+	ibctesting.ExpSimPassSend = false
+	ibctesting.ExpPassSend = false
+	err = suite.coordinator.ConnOpenTry(suite.chainB, suite.chainA, connB, connA)
+	suite.Require().Error(err)
+
+	// reset values
+	ibctesting.ExpSimPassSend = true
```

Comment:
this is a temp hack, we can update this to use a more robust method later, but since this is the first time we need messages to be allowed to fail I think it is okay for now